dask/array/core.py

```diff
@@ -817,7 +817,7 @@ def numblocks(self):
 
     @property
     def npartitions(self):
-        return np.prod(self.numblocks)
+        return reduce(mul, self.numblocks, 1)
```

Comment:
If it wasn't such a simple one liner then I would, but `reduce` with familiar arguments seems clearer IMO.